packages/contracts/test/voting/admin.ts

```diff
@@ -0,0 +1,202 @@
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
+import {expect} from 'chai';
+import {ethers} from 'hardhat';
+
+import {DAO} from '../../typechain';
+import {getMergedABI} from '../../utils/abi';
+import {findEvent} from '../../utils/event';
+import {customError, ERRORS} from '../test-utils/custom-error-helper';
+import {deployNewDAO} from '../test-utils/dao';
+import {getInterfaceID} from '../test-utils/interfaces';
+
+const EVENTS = {
+  ProposalCreated: 'ProposalCreated',
+  ProposalExecuted: 'ProposalExecuted',
+};
+
+// Permissions
+const EXECUTE_PROPOSAL_PERMISSION_ID = ethers.utils.id(
+  'EXECUTE_PROPOSAL_PERMISSION'
+);
+const EXECUTE_PERMISSION_ID = ethers.utils.id('EXECUTE_PERMISSION');
+
+describe('Admin plugin', function () {
+  let signers: SignerWithAddress[];
+  let plugin: any;
+  let dao: DAO;
+  let ownerAddress: string;
+  let dummyActions: any;
+  let dummyMetadata: string;
+
+  let mergedAbi: any;
+  let adminFactoryBytecode: any;
+
+  before(async () => {
+    signers = await ethers.getSigners();
+    ownerAddress = await signers[0].getAddress();
+
+    ({abi: mergedAbi, bytecode: adminFactoryBytecode} = await getMergedABI(
+      // @ts-ignore
+      hre,
+      'Admin',
+      ['DAO']
+    ));
+
+    dummyActions = [
+      {
```

Comment:
try use smock (already installed as a packet) as a target for the dummy action. then we know on an integration based testing if the actual DAO execution works or not.
Another idea would be to mock the DAO and check if `execute` got called by the plugin

Reply:
Thanks, amended